### PR TITLE
feat(api): handle local mode in airgap

### PR DESCRIPTION
**Pull request by UncleGedd**
## Description

- Ensures we can run the app locally in the airgap
- Refactors config logic into an `init` function

### CODE REVIEW

File: src/pkg/api/airgap.go [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+			os.Exit(1)
+		}
+	}()
+	<-stop
+	slog.Info("Shutting down server...")
+	return nil
+}
+
+// isAirgapped checks if we're in the airgap by checking Google and Cloudflare DNS servers
+func isAirgapped(timeout time.Duration) bool {
```
Comment (UncleGedd) [inline src/pkg/api/airgap.go:48]: TODO: dep inject the IPs to check so we can test properly

Comment (UncleGedd) [reply src/pkg/api/airgap.go:48]: also this function is gross 😅

Comment (UncleGedd) [reply src/pkg/api/airgap.go]: nvm, simplified this fn to the point it's not worth testing

**Comment by UncleGedd** [comment]
Chatted with @TristanHoladay , going to make the cert + domain configurable, this will:
- make testing easier
- allow customer configuration in `uds ui` to avoid the HTTPS errors when in the airgap

File: src/pkg/api/airgap.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	}
+	defer cleanupCerts()
+	stop := make(chan os.Signal, 1)
+	signal.Notify(stop, os.Interrupt, syscall.SIGTERM)
+	go func() {
+		//nolint:gosec,govet
+		err := http.ListenAndServeTLS(":8443", "airgap-cert.pem", "airgap-key.pem", r)
+		if err != nil {
+			slog.Error("Failed to start server:", "error", err)
+			os.Exit(1)
```
Comment (catsby) [inline src/pkg/api/airgap.go:39]: If we hit this `os.Exit(1)`, the deferred `cleanupCerts()` func will not run

File: src/pkg/api/start.go [hunk 3]
```diff
@@ -258,36 +258,40 @@ func fileServer(r chi.Router, root http.FileSystem) error {
...
-		// Create a server with TLS config
-		server := &http.Server{
-			Addr:      "127.0.0.1:8443",
-			Handler:   r,
-			TLSConfig: tlsConfig,
-		}
+	// connected to internet, create tls config from embedded cert and key
+	cert, err := tls.X509KeyPair(localCert, localKey)
+	if err != nil {
+		log.Fatalf("Failed to load embedded certificate: %v", err)
```
Comment (catsby) [inline src/pkg/api/start.go:279]: `log.Fatalf()` logs a message then invokes `os.Exit(1)`. We invoke `api.Serve()` from `main.go` ~line 33 where, where if `Serve()` returns an error we `os.Exit(1)` so we can simply log the error here and return it. This keeps up with using `slog` while passing control back to `main.go` 

```suggestion
		slog.Error("failed to load embedded certificate", "error", err)
		return err
```

File: src/pkg/api/airgap.go (hunk 2 above)
Comment (UncleGedd) [reply src/pkg/api/airgap.go:39]: fixed!

File: src/pkg/api/start.go [hunk 4]
```diff
@@ -258,36 +258,40 @@ func fileServer(r chi.Router, root http.FileSystem) error {
...
+		TLSConfig:         tlsConfig,
+		ReadHeaderTimeout: 10 * time.Second,
+	}
 
-		if err = server.ListenAndServeTLS("", ""); err != nil {
-			message.WarnErrf(err, "server failed to start: %s", err.Error())
-			return err
-		}
+	if err = server.ListenAndServeTLS("", ""); err != nil {
+		message.WarnErrf(err, "server failed to start: %s", err.Error())
```
[the remark this answers is not in the document]
Comment (UncleGedd) [reply src/pkg/api/start.go:294]: > If we're just watching the logs via another means (can we?), will this message show?

Today I don't think we handle the case of watching logs from another means. Noting that we have an issue (#432) to remove use of `message` entirely

**Review by decleaver — APPROVED** [review]
👍

**Review by catsby — APPROVED** [review]

**Pull request merged**